src/server/pfs/server/driver.go

```diff
@@ -632,6 +633,45 @@ func (d *driver) isPathModifiedInCommit(ctx context.Context, commit *pfs.Commit,
 	return found, nil
 }
 
+func (d *driver) getCompactedDiffFileSet(ctx context.Context, commit *pfs.Commit) (*fileset.ID, error) {
+	var compactedDiff *fileset.ID
+	return compactedDiff, backoff.RetryUntilCancel(ctx, func() error {
+		diff, err := d.getDiffFileSetIfCompacted(ctx, commit)
+		if err != nil {
+			return err
+		}
+		if diff != nil {
+			compactedDiff = diff
+			return nil
+		}
+		return errors.Errorf("diff file set for commit %s is not compacted yet", commit.ID)
+	}, backoff.NewConstantBackOff(time.Second), backoff.NotifyCtx(ctx, "getCompactedDiffFileSet for "+commit.ID))
+}
+
+func (d *driver) getDiffFileSetIfCompacted(ctx context.Context, commit *pfs.Commit) (*fileset.ID, error) {
```

Comment:
I think this code should just be a part of `getCompactedDiffFileSet`. I don't see us needing the no-op event separate from the backoff, and the nil, nil return seems like an anti-pattern.